testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -217,78 +477,48 @@ def setup_suite(suite_tag, work_dir, model_runtime, config_file, baseline_dir,
         # flush existing regression suite output file
         open(work_dir + '/manage_regression_suite.py.out', 'w').close()
 
+    regression_script_code = ""
+
+    if not args.local_parallel:
+        # Create regression suite run script
+        regression_script_name = '{}/{}.py'.format(work_dir, suite_name)
+        regression_script = open('{}'.format(regression_script_name), 'w')
+
+        regression_script_code = write_regression_script_testcase_data_top(work_dir)
+
     for child in suite_tag:
         # Process <test> tags within the test suite
         if child.tag == 'test':
-            process_test_setup(child, config_file, work_dir, model_runtime,
-                               regression_script, baseline_dir, verbose)
-
-    regression_script.write("print('TEST RUNTIMES:')\n")
-    regression_script.write("case_output = '/case_outputs/'\n")
-    regression_script.write("totaltime = 0\n")
-    regression_script.write("for _, _, files in os.walk(base_path + "
-                            "case_output):\n")
-    regression_script.write("    for afile in sorted(files):\n")
-    regression_script.write("        outputfile = base_path + case_output + "
-                            "afile\n")
-    regression_script.write("        runtime = np.ceil(float(subprocess."
-                            "check_output(\n"
-                            "                ['grep', 'real', outputfile])."
-                            "decode('utf-8').split('\\n')[-2].split()[1]))\n")
-    regression_script.write("        totaltime += runtime\n")
-    regression_script.write("        mins = int(np.floor(runtime/60.0))\n")
-    regression_script.write("        secs = int(np.ceil(runtime - mins*60))\n")
-    regression_script.write("        print('{:02d}:{:02d} {}'.format(mins, "
-                            "secs, afile))\n")
-    regression_script.write("mins = int(np.floor(totaltime/60.0))\n")
-    regression_script.write("secs = int(np.ceil(totaltime - mins*60))\n")
-    regression_script.write("print('Total runtime {:02d}:{:02d}'.format(mins, "
-                            "secs))\n")
-    regression_script.write("\n")
-
-    regression_script.write("if test_failed:\n")
-    regression_script.write("    sys.exit(1)\n")
-    regression_script.write("else:\n")
-    regression_script.write("    sys.exit(0)\n")
-    regression_script.close()
+            regression_script_code = process_test_setup(
+                child,
+                config_file,
+                work_dir,
+                model_runtime,
+                regression_script_code,
+                baseline_dir,
+                verbose)
+
+    if not args.local_parallel:
+        dev_null = open('/dev/null', 'a')
+        subprocess.check_call(['chmod',
+                               'a+x',
+                               '{}'.format(regression_script_name)],
+                              stdout=dev_null,
+                              stderr=dev_null)
+        dev_null.close()
```

Comment:
@jamilgafur, I think this `chmod` is needed for the parallel script, too.  Do you think that's not the case? How are you running the parallel script?

Reply:
python parallel....py

Reply:
You can do that but that's not what most of our users would expect.  We would typically do:
```
./parallel_....py
```

Reply:
gotcha